methods/enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def range_return_array(arr)
+    if self.class == Range
+      self
+    else
+      arr
+    end
+  end
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a
+    arr.size.times do |index|
+      yield arr[index]
+    end
+    range_return_array(arr)
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a
+    arr.size.times do |index|
+      yield arr[index], index
+    end
+    range_return_array(arr)
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = to_a
+    answer = []
+    arr.my_each do |item|
+      answer.push(item) if yield item
+    end
+    answer
+  end
+
+  def my_all?(arg = nil)
+    arr = to_a
+
+    if block_given?
+      arr.my_each { |item| return false unless yield item }
+      true
+    elsif !block_given? && !arg
+      arr.my_each { |item| return false unless item }
+      true
+    elsif arg.class == Class
+      arr.my_each { |item| return false unless item.is_a? arg }
+      true
+    elsif arg.class == Regexp
+      arr.my_each { |item| return false unless item.class =~ arg }
+      true
+    elsif arg.class != Regexp && arg.class != Class
+      arr.my_each { |item| return false unless item == arg }
+      true
+    end
+  end
```

Comment:
- [ ] when a Regex is passed as an argument it returns true if all of the collection matches the Regex

```Ruby
 words.my_all?(/d/)
```

expected `true`
            got `false`